client/gutenberg/extensions/wpcom-color-scheme/index.jsx

```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { registerBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getColorSchemesData from 'blocks/color-scheme-picker/constants';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormRadiosBar from 'components/forms/form-radios-bar';
+import 'blocks/color-scheme-picker/style.scss';
+
+class WpcomColorScheme extends Component {
+	state = {
+		colorSchemePreference: null,
+	};
+
+	componentDidMount() {
+		apiFetch( {
+			path: '/me/settings?http_envelope=1',
+		} ).then( response => {
+			this.setState( {
+				colorSchemePreference: response.calypso_preferences.colorScheme,
+			} );
+		} );
```

Comment:
noting here that because of the imperative nature of `apiFetch` we can't really replace the API calls in middleware without also making corresponding changes in the component. did we miss some opportunity in the middleware to transform the response?